src/Strategy/CommonClassesStrategy.php

```diff
@@ -135,4 +136,19 @@ public static function symfonyPsr18Instantiate()
     {
         return new SymfonyPsr18(null, Psr17FactoryDiscovery::findResponseFactory(), Psr17FactoryDiscovery::findStreamFactory());
     }
+
+    /**
+     * Can be use as a condition.
+     * @return bool
+     */
+    public static function isPsr17FactoryInstalled()
+    {
+        try {
+            Psr17FactoryDiscovery::findResponseFactory();
+        } catch (NotFoundException $e) {
```

Comment:
could there be other exceptions, e.g. when a class can't be loaded because missing interface or whatnot?

Reply:
maybe we should catch any throwable

Reply:
It can be other exceptions.. but are we interested in those? 
Maybe we should do a warning then